Enterprise 25.3 release branch (#874)
seqeralabs/docs: platform-enterprise_docs/studios/add-studio-git-repo.md

---

> title: "Import from Git repository"

Reviewer: So I would add a couple of paragraph one about the cloning behavior, and one about the docker builrs.

## Cloning

- The content of the git repository is cloned into the studio at first launch only, and it uses the same commit that was selected, or resolved while creating the studio in the first place.
- Cloning can be disabled, and the path at which the repository content will be cloned can be defined in the schema
- This configurability allows you to:
   + Clone disabled: you can share a public/private template just for the tool
   + Clone path configuration: can be used to configure specific functionalities of the tool with config files without the need to build a different docker image.

We decided not to store the `.git` folder, and not share the platform credentials with the studio for the time being. It should be clear to users that they won't have that repository linked to their studio once they are in, meaning:
- there are no preprovisioned git credentials available for them to use in the studio
- they cannot push/pull from the configured repository

> We might enhance/change this behavior in the future, but it's not yet possible due to the shared sutdios constraints.

## Docker builds

- The schema can define a `Dockerfile` that has to be inside the `.seqera` folder.
- Build context is limited to 10MB (same as wave)
- The studio image will be build by wave when the studio is added, and cached, meaning that if you create a completely separate studio, and the repo has not changed, platform will reuse the existing image without rebuild.
- Dockerfile based studios cannot be pushed to the wave community registry and so need some preconfiguration:
    1. The user has to configure a specific registry with an environment variable.
    2. Each workspace that wants to use the functionality needs to have some credentials available in the workspace to push to the defined registry.
    3. Usual limitations about private images (private for pull) apply, so the only supported registry/CE combination for a fully private dockerfile based studio) is ECR and AWS.

---
title: "Import from Git repository"
description: "Add a Studio in Platform."
date created: "2025-09-04"
last updated: "2025-11-14"
tags: [studio-git, git-repository, session, studios]
---

:::info[**Prerequisites**]
You will need the following to get started:

- **Maintain** role permissions (minimum)
- A compute environment with sufficient resources (scale based on data volume)
- [Data Explorer](../data/data-explorer) enabled
- Git credentials configured in your workspace
- A Git repository containing `seqera/studio-config.yaml`
:::

**Limitations**

- Compute environments are workspace-specific and cannot be defined in Git repositories. Select the compute environment when adding a Studio.
- Data links cannot be referenced in Git repositories. Select data links when adding a Studio.
- Git repositories with multiple Studio configurations are not supported. However, it is possible to use a Git repository with multiple branches and different configurations in each branch.

### Create the required configuration files 

**.seqera/studio-config.yaml configuration file**

> The schema can define a Dockerfile, which has to be inside the `.seqera` folder. Dockerfile-based Studios cannot be pushed to the Wave community registry and and the following applies:

Author: @t0randr, please can you review this content. Thank you!

---
title: "Import from Git repository"
description: "Add a Studio in Platform."
date created: "2025-09-04"
last updated: "2025-11-14"
tags: [studio-git, git-repository, session, studios]
---

:::info[**Prerequisites**]
You will need the following to get started:

- **Maintain** role permissions (minimum)
- A compute environment with sufficient resources (scale based on data volume)
- [Data Explorer](../data/data-explorer) enabled
- Git credentials configured in your workspace
- A Git repository containing a `seqera` folder
:::

**Limitations**

> - Compute environments are platform-specific and cannot be defined in external Git repositories. Select the compute environment when adding a Studio.

Reviewer 2: ```suggestion
- Compute environments are Platform-specific and cannot be defined in external Git repositories. Select the compute environment when you add a Studio.
```

    dependencies:
        condaEnvironmentFile: "environment.yaml" # Define additional libraries (and versions). Ignored for `dockerfile`
    computeRequirements:
        awsBatch:                                # Ignored for non-AWS batch CE
            cpu: 2                               # Number of CPUs to use. Defaults to 2
            gpu: 0                               # Number of GPUs to use (if the CE supports GPUs). Defaults to 0
            memory: 8192                         # Memory allocated in MiB. Defaults to 8192.
    environmentVariables:                        # Ordered sequence of elements that are objects (or mappings) of key-value pairs
        -   name: "var1"
            value: "value1"
        -   name: "var2"
            value: "value2"
    management:                                  # Session management settings
        lifespanHours: 1                         # Ignored if workspace lifespan is set
        isPrivate: false                         # Defaults to `false`
```

The schema can define a Dockerfile, which has to be inside the `.seqera` folder. The following limitations apply:

> - You need to configure a target repository using the `TOWER_DATA_STUDIO_WAVE_CUSTOM_IMAGE_REGISTRY` and `TOWER_DATA_STUDIO_WAVE_CUSTOM_IMAGE_REPOSITORY` environment variables. If no repository configuration is specified, the build will fail.

Reviewer 2: ```suggestion
- You must configure a target repository using the `TOWER_DATA_STUDIO_WAVE_CUSTOM_IMAGE_REGISTRY` and `TOWER_DATA_STUDIO_WAVE_CUSTOM_IMAGE_REPOSITORY` environment variables. If no repository configuration is specified, the build will fail.
```

- Each workspace needs to have credentials available in the workspace to push to the repository you've specified.

> - The only supported repository and compute environment combination for a fully private Dockerfile-based Studio is ECR and AWS.

Reviewer 2: Feels like this should be the first listed limitation as it's an infrastructure requirement
